File: Data Structures & Algorithms/longest-common-subsequence/submission-3.cpp
class Solution {
public:
    int solve(string text1 , string text2 , int i , int j){
        if(i == text1.size() || j == text2.size()){
            return 0;
        }

        int ans = 0;
        if(text1[i] == text2[j]){
            ans = 1 + solve(text1 , text2 , i+1 , j+1);
        }
        else{
            ans = max(solve(text1 , text2 , i+1 , j) , solve(text1 , text2 , i , j+1));
        }
        return ans;
        
    }
    int longestCommonSubsequence(string text1, string text2) {
        int ans = solve(text1 , text2 , 0 , 0);
        return ans;
    }
};

File: Data Structures & Algorithms/balanced-binary-tree/submission-3.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int solve(TreeNode root , boolean []ans){
        if(root == null){
            return 0;
        }

        int left = solve(root.left , ans);
        int right = solve(root.right , ans);
        
        if(Math.abs(left-right) > 1){
            ans[0] = false;
        }
        
        return 1 + Math.max(left , right);
    }
    public boolean isBalanced(TreeNode root) {
        boolean [] ans = {true};
        solve(root , ans);
        return ans[0];
    }
}
